### PR TITLE
codegen: Use target pointer size consistently for layout calculations

### LINKED ISSUE #1284: Wrong padding size generated from different arch
A generated code via cross compiling environment is difference from naitive generated code.

The cross compiling environment is x86_64 linux .
And a target environment is RaspberryPi(arm-unknown-linux-gnueabihf).

- bindgen 0.35.0
- llvm 3.9.1

### Input C/C++ Header

```C++
typedef int aligned[64] __attribute__((__aligned__(8)));
typedef int array[64];

typedef struct {
  aligned buf; // 256
  int x;
} a_record_t;

typedef struct
{
  array buf;
  int x;
} b_record_t;
```

### Bindgen Invocation

On x86_64 linux, I have generated code as gen_arm.rs, and send compiled binary to my raspberrypi.

```
x86_64 $ mkdir pi
x86_64 $ sshfs pi@raspberry.local:/usr pi
x86_64 $ TARGET=arm-unknown-linux-gnueabihf bindgen --output gen_arm.rs wrapper.h -- -I pi/include/clang/3.9.1/include --sysroot=/work/pi -I pi/include/arm-linux-gnueabihf
x86_64 $ rustc -C linker=armv6-rpi-linux-gnueabihf-gcc --target arm-unknown-linux-gnueabihf --test gen_arm.rs
x86_64 $ scp gen_arm pi@raspberry.local:.
```

On RaspberryPi, I have generated code ordinally and compile it using native rust compiler.

```
rpi $ bindgen --output gen_rpi.rs wrapper.h -- -I /usr/lib/clang/3.9.1/include --sysroot=/usr -I /usr/include/arm-linux-gnueabihf
rpi $ rustc --test gen_rpi.rs
```

### Actual Results

Below is the result of the layout test which is generated on x86_64 linux environment.

```
rpi $ ./gen_arm
running 2 tests
test bindgen_test_layout_b_record_t ... ok
test bindgen_test_layout_a_record_t ... FAILED

failures:

---- bindgen_test_layout_a_record_t stdout ----
        thread 'bindgen_test_layout_a_record_t' panicked at 'assertion failed: `(left == right)`
  left: `260`,
 right: `264`: Size of: a_record_t', gen_arm.rs:13:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.


failures:
    bindgen_test_layout_a_record_t

test result: FAILED. 1 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out
```

On the other hand, below is the result of the same test above.
But the test code have been generated on RaspberryPi.

```
rpi $ ./gen_rpi
running 2 tests
test bindgen_test_layout_b_record_t ... ok
test bindgen_test_layout_a_record_t ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```

### Expected Results

What is expected of results of tests, both results are matching.

```
rpi $ ./gen_arm
running 2 tests
test bindgen_test_layout_b_record_t ... ok
test bindgen_test_layout_a_record_t ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```

### PR DESCRIPTION
**Pull request by emilio**
Assuming new enough libclang, this PR makes cross-compilation do layout calculations properly.

Fixes #1284

### CODE REVIEW

File: src/clang.rs [hunk 1]
```diff
@@ -1812,3 +1812,35 @@ impl Drop for EvalResult {
...
+impl TargetInfo {
+    /// Tries to obtain target information from libclang.
+    pub fn new(tu: &TranslationUnit) -> Option<Self> {
+        if !clang_getTranslationUnitTargetInfo::is_loaded() {
+            return None;
+        }
+        let triple;
+        let pointer_width;
+        unsafe {
+            let ti = clang_getTranslationUnitTargetInfo(tu.x);
```
Comment (fitzgen) [inline src/clang.rs]: Oh, nice -- this is the API you sent patches upstream to add, right?

File: src/codegen/struct_layout.rs [hunk 2]
```diff
@@ -165,15 +164,13 @@ impl<'a> StructLayoutTracker<'a> {
...
-            if let Some(layout) = self.ctx.resolve_type(inner).layout(
-                self.ctx,
-            )
-            {
-                if layout.align > mem::size_of::<*mut ()>() {
-                    field_layout.size = align_to(layout.size, layout.align) *
-                        len;
-                    field_layout.align = mem::size_of::<*mut ()>();
+            // access, since they won't be properly aligned, but there's not too
+            // much we can do about it.
```
Comment (fitzgen) [inline src/codegen/struct_layout.rs]: For posterity, we can use `#[repr(align(..))]` to fix this ultra hack now (given a new enough target rust).

File: ci/before_install.sh [hunk 3]
```diff
@@ -1,36 +1,60 @@
-set -ex
+set -e
```
Comment (fitzgen) [inline ci/before_install.sh]: I think having `-x` is useful for debugging CI runs -- is there a particular reason you removed it?

File: src/clang.rs (hunk 1 above)
Comment (emilio) [reply src/clang.rs]: Right :)

File: ci/before_install.sh (hunk 3 above)
Comment (emilio) [reply ci/before_install.sh]: Not really, the script was taken from clang-sys, and they removed it upstream. But I'll re-add.

**Pull request merged**